codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -272,10 +272,10 @@ private boolean allowNotificationEvent() {
         public ChannelPromise setSuccess(Void result) {
             if (allowNotificationEvent()) {
                 ++successfulCount;
-                if (successfulCount == expectedCount && doneAllocating) {
-                    promise.setSuccess(result);
-                    return super.setSuccess(result);
-                }
+            }
+            if (successfulCount == expectedCount && doneAllocating && !isDone()) {
```

Comment:
@Scottmitch see [my previous comment](https://github.com/netty/netty/pull/3725#discussion_r29676013). I followed this in the debugger and saw that `doneAllocatingPromises` is actually calling `this` version of `setSuccess`.  The "real" promise completes because `doneAllocatingPromises` sets it before calling `this.setSuccess`, which ends up doing nothing ... but we never noticed because nothing ever checks `isDone`.  I stumbled on this while playing around with a gRPC unit test.